docs/2-building-blocks-of-oop-part-1/hometask/src/Triangle.ts

```diff
@@ -1,3 +1,38 @@
-import { Shape } from './Shape';
+import { Shape } from './Shape'
+import { Point } from './Point';
 
-export class Triangle extends Shape {}
+export class Triangle extends Shape {
+  private triangleType: string;
+
+  constructor(pointA: Point, pointB: Point, pointC: Point);
+  constructor(pointA: Point, pointB: Point, pointC: Point, color: string, filled: boolean);
+  constructor(pointA: Point, pointB: Point, pointC: Point, color?: string, filled?: boolean) {
+    if (typeof color === 'string' && typeof filled === 'boolean') {
+      super([pointA, pointB, pointC], color, filled);
+    } else {
+      super([pointA, pointB, pointC]);
+    }
+    this.identifyTriangleType();
+  }
+
+  identifyTriangleType() {
```

Comment:
Hi @siyuejiwang , it seems like a private method here, better use `private` keyword and `_{MethodName}` or `__{MethodName}`(based on your coding style) to tell others what it is.